pandas/plotting/_core.py

```diff
@@ -1805,6 +1805,10 @@ def _plot(data, x=None, y=None, subplots=False,
                         except (IndexError, KeyError, TypeError):
                             pass
 
+                if type(y) == type(x):
```

Comment:
> I had to add the type check because otherwise it failed a check where y is a numpy array.

Could you clarify what you mean by this comment?  Potentially we should add that as a comment to this conditional.  For the purpose of this GitHub discussion, an example would be good too.

Reply:
Will do. I actually would like to change this to handle the case where `y` is a list. I will send an updated pull request. I can put together an example.